api/types/client.go

```diff
@@ -319,4 +321,54 @@ type VanClientInterface interface {
 	GetIngressDefault() string
 	RevokeAccess(ctx context.Context) error
 	NetworkStatus() ([]*SiteInfo, error)
+	ServiceManager(namespace string) Services
+	ConfigMapManager(namespace string) ConfigMaps
+	DeploymentManager(namespace string) Deployments
+	SecretManager(namespace string) Secrets
+	StatefulSetManager(namespace string) StatefulSets
+}
+
+type Services interface {
+	GetService(name string, options *v1.GetOptions) (*corev1.Service, bool, error)
```

Comment:
I have been trying to reduce dependencies on the client package, particularly from within the pkg tree. I would personally prefer to see these new interfaces introduced elsewhere.

I have also been trying to contain dependence on the kubernetes API a little better. This PR takes things in a different direction, namely making kubernetes resource a central aspect of the entire project. That is not necessarily wrong, but it does give me some concern. Should StatefulSets really be so central a concept?

Reply:
> I have been trying to reduce dependencies on the client package, particularly from within the pkg tree. I would personally prefer to see these new interfaces introduced elsewhere.

I agree that `pkg` tree should not depend on the client package and I have just moved the
Kubernetes implementation of the new interfaces into `pkg/kube`.

> I have also been trying to contain dependence on the kubernetes API a little better. This PR takes things in a different direction, namely making kubernetes resource a central aspect of the entire project. That is not necessarily wrong, but it does give me some concern.

The idea is to use the kubernetes models as an internal way to represent Skupper resources. I made it this way, because almost everything is tied to ConfigMaps, Secrets and Deployments (mainly). It sounded to me like a cleaner/safer approach to promote reuse of existing logic.

As an example, decoupling the existing `VanClientInterface` implementation from the kubernetes models, sounds to me like a bigger risk than simply providing appropriate mechanisms to store and manipulate the kubernetes resources that are already broadly manipulated by CLI and by the Controller.

> Should StatefulSets really be so central a concept?

I have added it mainly because of the existing logic that deals with headless services, but actually I am not sure it will be needed in different platforms.

Same thing for DaemonSets. I have not added generic CRUD interface to manipulate DaemonSets.
Do you think it might be needed?